src/gsUtils/gsUtils.h

```diff
@@ -138,6 +138,36 @@ inline void string_replace(std::string& str,
     }
 }
 
+/// \brief Splits a string \a str into substrings with \a ch as separator.
+inline std::vector<std::string> split(std::string str, char ch)
+{
+    std::vector<std::string> result;
+    size_t pos = 0;
+    while ((pos = str.find(ch)) != std::string::npos)
+    {
+        if (pos == 0)
+        {
+            str.erase(0, 1);
+            continue;
+        }
+        result.push_back(str.substr(0, pos));
+        str.erase(0, pos + 1);
+    }
+    if(str.size())
+        result.push_back(str);
+    return result;
+}
```

Comment:
erease is an expensive operation since it requires a copy of the remainder of the string. Thus, this algorithm is O(N^2).
```

{
    std::vector<std::string> result;
    size_t begin = 0;
    size_t end;
    while ((end = str.find(ch, begin)) != std::string::npos)
    {
        if (end == begin)
        {
            ++begin;
            continue;
        }
        result.push_back(str.substr(begin, end-begin));
        begin = end + 1;
    }
    end = str.size();
    if (begin<end)
        result.push_back(str.substr(begin, end-begin));
    return result;
}
```

I this should be correct (needs testing)
For sure, its only O(N).

Reply:
Done.
Also as we discussed yesterday.
`template<typename T> inline T convert_to(const std::string&);` should work too - hope so.